server/app/services/docker/service_creator.rb

```diff
@@ -173,6 +175,46 @@ def build_networks
       networks
     end
 
+    def build_volumes(instance_number)
+      volume_specs = []
+      grid_service.volumes.each do |vol|
+        vol_string = remove_volume_flags(vol.dup)
```

Comment:
I think all this complexity can be avoided by just matching what the agent does...

```ruby
next unless vol.include?(':') # anonymous volume
host_vol, container_path, flags = vol.split(':') # flags is optional
next if host_vol.start_with?('/') # bind mount from host
```